app/models/ability.rb

```diff
@@ -141,6 +141,30 @@ def initialize(user)
         ct.clonable?
       end
 
+      can :list_user_emails, Course do |c|
+        can? :teach, c
+      end
+
+      can :send_mail_to_participants, Course do |c|
+        can? :teach, c
+      end
+
+      can :manage_deadlines, Course do |c|
+        can? :teach, c
+      end
+
+      can :manage_unlocks, Course do |c|
+        can? :teach, c
+      end
+
+      can :manage_exercises, Course do |c|
+        can? :teach, c.organization
+      end
+
+      can :edit_course_paramaters, Course do |c|
+        can? :teach, c.organization
+      end
```

Comment:
I'm wondering if these should be ordered by some logic, alphabetically or maybe grouped by logical uses